Develop pluu

Pluu opened the pull request:
## 작업 내역

**Project Configuration**
- Add, Kotlin : 1.1.3-2
- Add, kotlin-android-extensions
- Add, kotlin-kapt
- Add, Databinding AndroidProcessor
- Update, dependencies : compile -> implementation
- Update, com.android.tools.build:gradle : 3.0.0-alpha3 -> 3.0.0-alpha7
- Update, buildToolsVersion : 25.0.2 -> 25.0.3
- Update, ArchVersion : 1.0.0-alpha1 -> 1.0.0-alpha5

**Kotlin File**
- Application File
- Network, Module + Service + Entity
- DB, Database/Dao/Entity
- ImageBindingAdapter.java
- Convert
  - Resource.java
  - Status.java 

**Modify**
- API 가 존재하지않을 경우 예외처리 추가

Pluu committed:
Add Image Resource

- ic_language_green_24dp
- ic_stars_green_24dp

Pluu committed:
Configure in Kotlin Project

- Add, Kotlin : 1.1.3-2
- Add, kotlin-android-extensions
- Update, com.android.tools.build:gradle : 3.0.0-alpha3 -> 3.0.0-alpha7
- Update, buildToolsVersion : 25.0.2 -> 25.0.3
- Update, dependencies : compile -> implementation

Pluu committed:
Configure in Kotlin Project

- Add, Databinding AndroidProcessor
- Update, ArchVersion : 1.0.0-alpha1 -> 1.0.0-alpha5
- kapt Convert

Pluu committed:
Kotlin Convert
- Application

Pluu committed:
Rename, Gradle Wrapper Update

Pluu committed:
Retrofit 관련 Kotlin 변환

Pluu committed:
DB 관련 Kotlin Convert

Pluu committed:
Develop

- Convert
  - Resource.java
  - Status.java
  - ImageBindingAdapter.java

Pluu committed:
API 키가 존재하지않을 경우 처리

- NetworkBoundResource#saveCallResult 을 @NonNull을 @Nullable 으로 처리

Pluu commented:
2회 모임 이후 커밋 스탬프

Pluu committed:
Deploy

- ignore Update
- API Key Binding
- Convert, ListBindingAdapter
- Convert, MovieListAdapter
- Convert, BaseAdapter

cwdoh commented on app/src/main/java/iammert/com/androidarchitecture/AAApp.kt:
```diff
@@ -0,0 +1,32 @@
+                .inject(this)
+    }
+
+    override fun activityInjector() = activityDispatchingInjector
```
memo: DI에 의해 리턴값의 명시를 하지 않는 경우에 대한 논의

- 외부에 제공하지 않는 경우에는 명시하여 산출 도큐먼트 등에 대한 모호성을 없애는게 좋겠음
- 내부에서만 사용되는 코드의 경우 약간 모호하게 보이더라도 굳이 명시를 하지 않게 간결하게 유지

cwdoh commented on app/src/main/java/iammert/com/androidarchitecture/databinding/ImageBindingAdapter.kt:
```diff
@@ -0,0 +1,22 @@
+ */
+@BindingAdapter(value = "url")
+fun loadImageUrl(view: ImageView, url: String?) {
+    if (!url.isNullOrEmpty()) {
```
memo: 유사한 유틸리티가 몇가지 더 있으니 살펴보고 사용하면 도움이 될 듯

Pluu committed:
Develop

- android_plugin_version : 3.0.0-alpha7 > 3.0.0-alpha8
- gradle-4.1-milestone-1 -> gradle-4.1-rc-1
- Convert, ApiConstants
- Convert, NetworkBoundResource
- Modify, Resource

Pluu committed:
Develop

- Dagger Module
- Modify, NetworkBoundResource

Pluu committed:
Develop

- Convert, Main Package
- Convert, ViewModel

Pluu committed:
Convert

Pluu committed:
FirstOrNull 써보기

Pluu committed:
범준 & Pluu 컨플릿 합체
